crates/proof-of-sql/src/sql/util.rs

```diff
@@ -13,60 +13,61 @@ use sqlparser::ast::BinaryOperator;
 ///
 /// # Returns
 ///
-/// * `true` if the operation is valid, `false` otherwise.
+/// * `Some(result_type)` if the operation is valid, `None` otherwise.
 pub(crate) fn type_check_binary_operation(
```

Comment:
Maybe this function should be renamed to something like `binary_operation_column_types` since it returns a `Option<ColumnType>`?